os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -0,0 +1,763 @@
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_aes_encrypt(hal_data *dec_data, hal_aes_param *aes_param, uint32_t key_idx, hal_data *enc_data)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = dec_data;
+	in_output.output_data = enc_data;
+
+	ret = ameba_hal_aes_encrypt(&in_output, aes_param, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_aes_decrypt(hal_data *enc_data, hal_aes_param *aes_param, uint32_t key_idx, hal_data *dec_data)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = enc_data;
+	in_output.output_data = dec_data;
+
+	ret = ameba_hal_aes_decrypt(&in_output, aes_param, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_rsa_encrypt(hal_data *dec_data, hal_rsa_mode *rsa_mode, uint32_t key_idx, hal_data *enc_data)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = dec_data;
+	in_output.output_data = enc_data;
+
+	ret = ameba_hal_rsa_encrypt(&in_output, rsa_mode, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_rsa_decrypt(hal_data *enc_data, hal_rsa_mode *rsa_mode, uint32_t key_idx, hal_data *dec_data)
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = enc_data;
+	in_output.output_data = dec_data;
+	ret = ameba_hal_rsa_decrypt(&in_output, rsa_mode, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_write_storage(uint32_t ss_idx, hal_data *data)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	ret = ameba_hal_write_storage(ss_idx, data);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_read_storage(uint32_t ss_idx, hal_data *data)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	ret = ameba_hal_read_storage(ss_idx, data);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_delete_storage(uint32_t ss_idx)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	ret = ameba_hal_delete_storage(ss_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+
+static struct sec_ops_s g_ameba_ops = {
```

Comment:
How about naming `g_ameba_ops` to `g_ameba_ops_s` as Taejun's suggestion?

Reply:
Okay, I will modify accordingly.